codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -208,6 +209,9 @@ protected void encode(ChannelHandlerContext ctx, HttpObject msg, List<Object> ou
                 ensureContent(msg);
                 if (encodeContent((HttpContent) msg, out)) {
                     state = State.AWAIT_HEADERS;
+                } else if (out.isEmpty()) {
+                    // MessageToMessageCodec needs at least one output message
+                    out.add(new DefaultHttpContent(Unpooled.EMPTY_BUFFER));
```

Comment:
I think we should better move this to `encodeContent` WDYT ?

Reply:
@yawkat wdyt ?

Reply:
sorry didn't have time to try this yesterday.

I think it makes more sense to have this in encode. It's only necessary if no further items are added after the call to encodeContent, and in encode, it's more clear that this is the case. In both branches where encodeContent is called (AWAIT_CONTENT and AWAIT_HEADERS + FullHttpResponse) it turns out that encodeContent is only called once and is the last one to add to `out` as well, so moving the patch would work, but imo this is much harder to see so the check makes more sense in encode.